C Error on declaring value on a value

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

**V version:** 0.1.29 96b7cce
**OS:** Linux

**What did you do?**
```
1 := 1
"a" := 1
```

**What did you expect to see?**
V Compiler error:
```
test.v:1:1: error: unexpected expression `v.ast.IntegerLiteral`
    1 | 1 := 1
      | ^
    2 | "a" := 1
test.v:2:1: error: unexpected expression `v.ast.StringLiteral`
    1 | 1 := 1
    2 | "a" := 1
      | ~~~
```
 
**What did you see instead?**
```
==================
/tmp/v/test.tmp.c: In function ‘main__main’:
/tmp/v/test.tmp.c:7861:6: error: expected identifier or ‘(’ before numeric constant
 7861 |  int 1 = 1;
      |      ^
/tmp/v/test.tmp.c:7862:14: error: expected declaration specifiers or ‘...’ before string constant
 7862 |  int tos_lit("a") = 1;
      |              ^~~
...
==================
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1846,7 +1846,7 @@ pub fn (mut c Checker) assign_stmt(mut assign_stmt ast.AssignStmt) {
 			}
 			else {
 				if is_decl {
-					c.error('unexpected expression `${typeof(left)}`', left.position())
+					c.error('non-name `$left` on left side of `:=`', left.position())
 				}
 			}
 		}

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1844,7 +1844,11 @@ pub fn (mut c Checker) assign_stmt(mut assign_stmt ast.AssignStmt) {
 						assign_stmt.pos)
 				}
 			}
-			else {}
+			else {
+				if is_decl {
+					c.error('unexpected expression `${typeof(left)}`', left.position())
+				}
+			}
 		}
 		left_type_unwrapped := c.unwrap_generic(left_type)
 		right_type_unwrapped := c.unwrap_generic(right_type)

